src/main/java/com/fasterxml/jackson/core/json/UTF8StreamJsonParser.java

```diff
@@ -2544,7 +2544,14 @@ private final void _finishString2(char[] outBuf, int outPtr)
                     outBuf = _textBuffer.finishCurrentSegment();
                     outPtr = 0;
                 }
-                final int max = Math.min(_inputEnd, (ptr + (outBuf.length - outPtr)));
+
+                int max;
+                try {
+                    max = Math.min(_inputEnd, Math.addExact(ptr, outBuf.length - outPtr));
+                } catch (ArithmeticException ex) {
+                    max = Integer.MAX_VALUE;
```

Comment:
eg would `_inputEnd` work?

Reply:
min of both?

Reply:
actually, looking at the code Integer.MAX_VALUE looks like a good fallback

Reply:
Okay, I think Integer.MAX_VALUE would be fine since we're just trying to prevent an overflow here. `max` would generally be larger than the max string length or `_inputEnd`. Verification that we don't go beyond the max string length happens when data is read.